soroban-test-wasms/wasm-workspace/loadgen/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+#![no_std]
+use soroban_sdk::{contract, contractimpl, Bytes, Env, U256};
+
+#[contract]
+pub struct Contract;
+
+
+const EXTEND_LEDGER: u32 = 5_000_000;
+
+#[contractimpl]
+impl Contract {
+
+    // This will write num_entries of size size_kilo_bytes. The initial entry will have a key of
+    // starting_index, the next key starting_index + 1...
+    pub fn write(e: Env, starting_index: u32, num_entries: u32, size_kilo_bytes: u32) {
+        if num_entries == 0 || size_kilo_bytes == 0{
+            panic!("num_entries and size_kilo_bytes must be greater than 0");
+        }
+
+        let slice = [0_u8; 1024];
+        let mut bytes = Bytes::new(&e);
+        for _ in 0..size_kilo_bytes {
+            bytes.extend_from_slice(&slice);
+        }
+
+        for i in starting_index..starting_index + num_entries {
+            e.storage().persistent().set(&i, &bytes);
+            e.storage().persistent().extend_ttl(&i, EXTEND_LEDGER, EXTEND_LEDGER);
+        }
+    }
+
+    pub fn do_work(e: Env, guest_cycles: u64, host_cycles: u64)
+    {
+        // We just want to do a lot of math, overflow is fine
+        let mut val = 1.3;
+        for _ in 0..guest_cycles {
+            val = val * 1.15;
```

Comment:
Have you actually somehow managed to get this working? Floating point math should be prohibited (you shouldn't even be able to upload this - let me know if you could). Also overflows will cause panics, so that will fail as well. Given that these are guest ops, there shouldn't be difference between addition and multiplication, so you can just add loop counter to the integer (I'd hope compiler isn't smart enough to optimize that).

Reply:
Hmm, I've been testing with the `InvokeFostFunctionTests` framework and it seems to work, as in a call to `do_work` succeeds, and if I increase the cycle count if fails due to resource limit being exceeded. Maybe we skip some checks in the unit tests? I haven't tried actually submitting it to testnet.